pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -406,8 +407,32 @@ func newBlackholeFilter(enableAny bool) *listener.Filter {
 	return filter
 }
 
+func createRBACfilter(protocol plugin.ListenerProtocol, env *model.Environment, node *model.Proxy, push *model.PushContext) interface{} {
+	in := &plugin.InputParams{
+		Env:              env,
+		Node:             node,
+		Push:             push,
+		ListenerProtocol: protocol,
+	}
+	mutable := &plugin.MutableObjects{
+		FilterChains: []plugin.FilterChain{
+			{
+				ListenerProtocol: protocol,
+			},
+		},
+	}
+	if err := authz.NewPlugin().OnInboundListener(in, mutable); err == nil {
```

Comment:
this still doesn't feel right.. shouldn't we just add `OnInboundPasssthrough` or something similar?

Even if we are calling the plugin here, this still breaks the plugin model. Core pilot code should not know about plugins, it should just have a list of plugins the user turns on and call callbacks for them. Authz may not even be enabled but we are using the authz plugin regardless

Reply:
I think it could be a long term solution to add something like `OnInboundPasssthrough`. But I'm afraid this needs more design discussions, are we okay that this new plugin interface is only used for the passthrough filter chain? That sounds too limited use case and seems we're adding it only for solving the problem here. 

Instead, we may want to add it in a more general way that could better work for other use cases as well, this may need more design.   cc @lambdai 

Here I'm trying to do a short-term fix to only insert the authz filter in the new code path because authz is the only filter that needs to be inserted here, and I also want to avoid any side-effect of adding other filters from other plugins.

The authz plugin is enabled by default, I'm not sure in what case the customer would want to disable the plugin, they could just remove all their authz policies for the same effect. If we're really worried it may be disabled by some users, i can do some refactor to pass around the enabled plugin lists but this will require some more code changes in the other places in pilot.

Reply:
I don't think there is any point in having plugins at all if we are just going to call them directly from core pilot code. The whole intent of them is to keep separate concerns isolated. The other plugins can just do nothing `OnInboundPasssthrough`, so I don't think that is a concern.

Reply:
@howardjohn Cool, updated to use `OnInboundPasssthrough`, PTAL, thanks!